foundationdb/src/database.rs

```diff
@@ -64,18 +65,20 @@ impl Database {
     /// It might retry indefinitely if the transaction is highly contentious. It is recommended to
     /// set `TransactionOption::RetryLimit` or `TransactionOption::SetTimeout` on the transaction
     /// if the task need to be guaranteed to finish.
-    pub fn transact<F, Fut, Item>(&self, f: F) -> Box<Future<Item = Fut::Item, Error = FdbError>>
+    pub fn transact<F, Fut, Item, E>(&self, f: F) -> Box<Future<Item = Fut::Item, Error = Error>>
     where
         F: FnMut(Transaction) -> Fut + 'static,
-        Fut: Future<Item = Item, Error = FdbError> + 'static,
+        Fut: Future<Item = Item, Error = E> + 'static,
         Item: 'static,
+        E: Into<Error>,
```

Comment:
How about using `E: From<FdbError>` and use `E::From(fdb_err)` for internal error case? If caller is not emitting error on `F` they can use it with `FdbError`, and if caller uses custom error type with either `failure` or `error-chain`, they can add `From<FdbError>` for there own error type without much effort.